crates/cheatnet/tests/starknet/forking.rs

```diff
@@ -0,0 +1,213 @@
+use crate::common::deploy_contract;
+use crate::common::state::create_cheatnet_fork_state;
+use crate::{assert_error, assert_success};
+use blockifier::state::errors::StateError;
+use cairo_felt::Felt252;
+use camino::Utf8PathBuf;
+use cheatnet::cheatcodes::{CheatcodeError, EnhancedHintError};
+use cheatnet::constants::build_testing_state;
+use cheatnet::conversions::{
+    class_hash_from_felt, contract_address_from_felt, felt_selector_from_name,
+};
+use cheatnet::forking::state::ForkStateReader;
+use cheatnet::forking::worker::Worker;
+use cheatnet::rpc::call_contract;
+use cheatnet::state::CustomStateReader;
+use cheatnet::CheatnetState;
+use num_bigint::BigUint;
+use starknet::core::types::BlockId;
+use std::str::FromStr;
+
+#[test]
+fn fork_simple() {
+    let mut state = create_cheatnet_fork_state();
+
+    let contract_address = contract_address_from_felt(&Felt252::from(
+        BigUint::from_str(
+            "3216637956526895219277698311134811322769343974163380838558193911733621219342",
```

Comment:
What address is it? Does calling `increase_balance` not mutate the state? Won't it cause the test to fail when a new block is minted?

Reply:
This is the `HelloStarknet` contract address deployed by me for this specific test suite.
`increase_balance` mutates the state but only locally so it will not affect future runs

Reply:
Oh true, forgot we only mutate it locally, thanks!